File: app/src/main/java/igrek/forceawaken/domain/task/AnswerAgainTask.java
package igrek.forceawaken.domain.task;

import android.annotation.SuppressLint;

import javax.inject.Inject;

import igrek.forceawaken.AwakenActivity;
import igrek.forceawaken.dagger.DaggerIOC;
import igrek.forceawaken.service.ui.info.UserInfoService;

public class AnswerAgainTask implements AwakeTask {
	
	@Inject
	UserInfoService userInfoService;
	
	public AnswerAgainTask() {
		DaggerIOC.getFactoryComponent().inject(this);
	}
	
	@Override
	public AwakeTask getInstance() {
		return new AnswerAgainTask();
	}
	
	@Override
	public double getProbabilityWeight() {
		return 0; // TODO turn it on
	}
	
	@SuppressLint("NewApi")
	@Override
	public void run(AwakenActivity activity) {
		userInfoService.showInfoBar("Once again quest.");
		activity.startAlarmPlaying(0);
	}
}

File: app/src/main/java/igrek/forceawaken/task/AnswerAgainTask.kt
package igrek.forceawaken.task

import igrek.forceawaken.activity.AwakenActivity
import igrek.forceawaken.activity.AwakenActivityLayout
import igrek.forceawaken.info.UiInfoService
import igrek.forceawaken.inject.LazyExtractor
import igrek.forceawaken.inject.LazyInject
import igrek.forceawaken.inject.appFactory

class AnswerAgainTask(
        uiInfoService: LazyInject<UiInfoService> = appFactory.uiInfoService,
        awakenActivityLayout: LazyInject<AwakenActivityLayout> = appFactory.awakenActivityLayout,
) : AwakeTask {
    private val uiInfoService by LazyExtractor(uiInfoService)
    private val awakenActivityLayout by LazyExtractor(awakenActivityLayout)

    override val instance: AwakeTask
        get() = AnswerAgainTask()
    override val probabilityWeight: Double
        get() = 0.0

    override fun run(activity: AwakenActivity) {
        uiInfoService.showSnackbar("Once again quest.")
        awakenActivityLayout.startAlarmPlaying(0.0, System.currentTimeMillis())
    }
}